.pre-commit-config.yaml

```diff
@@ -0,0 +1,24 @@
+# Base Configuration
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.1.0
+  hooks:
+  - id: check-yaml
+  - id: check-json
+  - id: check-added-large-files
+  - id: pretty-format-json
+    args: [--autofix, --no-sort-keys]
+  - id: end-of-file-fixer
```

Comment:
Does this require that we have individual configurations for each rule we want to apply?

Asking because that strikes me as expensive to maintain, and hard to keep in sync with a codestyle.xml